authentication/account/service/user_blackbox_test.go

```diff
@@ -177,26 +188,44 @@ func (s *userServiceBlackboxTestSuite) TestListUsersToNotifyBeforeDeactivation()
 		config.GetUserDeactivationInactivityNotificationPeriodDaysFunc = func() time.Duration {
 			return 30 * 24 * time.Hour // 30 days
 		}
+		var msgToSend []notification.Message
 		notificationServiceMock := servicemock.NewNotificationServiceMock(s.T())
 		notificationServiceMock.SendMessageAsyncFunc = func(ctx context.Context, msg notification.Message, options ...rest.HTTPClientOption) (r chan error, r1 error) {
+			msgToSend = append(msgToSend, msg)
 			return nil, nil
 		}
 		userSvc := userservice.NewUserService(factory.NewServiceContext(s.Application, s.Application, nil, nil, factory.WithNotificationService(notificationServiceMock)), config)
 		// when
-		result, err := userSvc.NotifyIdentitiesBeforeDeactivation(ctx, time.Now)
+		result, err := userSvc.NotifyIdentitiesBeforeDeactivation(ctx, nowf)
 		// then
 		require.NoError(s.T(), err)
 		require.Len(s.T(), result, 2)
 		assert.Equal(s.T(), identity2.ID, result[0].ID)
 		assert.Equal(s.T(), identity1.ID, result[1].ID)
 		assert.Equal(s.T(), uint64(2), notificationServiceMock.SendMessageAsyncCounter)
 		// also check that the `DeactivationNotification` fields were set for both identities in the DB
-		for _, id := range []uuid.UUID{identity1.ID, identity2.ID} {
+		expiryDate := userservice.GetExpiryDate(config, nowf)
+		customs := []map[string]interface{}{}
+		for i, id := range []uuid.UUID{identity1.ID, identity2.ID} {
 			identity, err := s.Application.Identities().Load(ctx, id)
 			require.NoError(s.T(), err)
 			require.NotNil(s.T(), identity.DeactivationNotification)
 			assert.True(s.T(), time.Now().Sub(*identity.DeactivationNotification) < time.Second*2)
+			// also verify that the message to send to the user has the correct data
+			assert.Equal(s.T(), identity.ID.String(), msgToSend[i].TargetID)
```

Comment:
if we can't guarantee order then we can't make this assertion too.
`assert.Equal(s.T(), identity.ID.String(), msgToSend[i].TargetID)`
Because msg can be added any order to msgToSend.